### PR TITLE
Move Map.FindParentRoom to Room class, add Room.Get(RelativePosition)

### PR DESCRIPTION
**Pull request by Thundermaker300**

### CODE REVIEW

File: Exiled.API/Features/Map.cs [hunk 1]
```diff
@@ -186,6 +186,22 @@ public static Room FindParentRoom(GameObject objectInRoom)
             return room ?? Room.Get(objectInRoom.transform.position) ?? default;
         }
 
+        /// <summary>
+        /// Tries to find a room at the given <see cref="Vector3">position</see> using raycasting.
+        /// </summary>
+        /// <param name="position">The position to look for a room.</param>
+        /// <returns>The <see cref="Room"/> that the <see cref="Vector3"/> is located inside. Can be <see langword="null"/>.</returns>
+        public static Room FindParentRoom(Vector3 position)
+            => Room.Get(position);
```
Comment (iopietro) [inline Exiled.API/Features/Map.cs:195]:
```suggestion
        public static Room FindParentRoom(Vector3 position) => Room.Get(position);
```

File: Exiled.API/Features/Map.cs [hunk 2]
```diff
@@ -186,6 +186,22 @@ public static Room FindParentRoom(GameObject objectInRoom)
...
+
+        /// <summary>
+        /// Tries to find a room at the given <see cref="RelativePosition">position</see> using raycasting.
+        /// </summary>
+        /// <param name="position">The position to look for a room.</param>
+        /// <returns>The <see cref="Room"/> that the <see cref="RelativePosition"/> is located inside. Can be <see langword="null"/>.</returns>
+        public static Room FindParentRoom(RelativePosition position)
+            => Room.Get(position.Position);
```
Comment (iopietro) [inline Exiled.API/Features/Map.cs:203]:
```suggestion
        public static Room FindParentRoom(RelativePosition position) => Room.Get(position.Position);
```

**Review by iopietro — CHANGES_REQUESTED** [review]

**Review by iopietro — APPROVED** [review]

**Comment by louis1706** [comment]
> I'm wondering if we should rename these to `FindRoom`, as `FindParentRoom` is kind of misleading for Vector3s that don't have a "parent" room.
your  FindRoom() are useless here they should be all In Room api (also Map.FindParentRoom because it's a Room so should be in Room)

**Comment by Thundermaker300** [comment]
Moving `Map.FindParentRoom` is a breaking change. I added these because Room.Get() isn't exactly as obvious. Having them doesn't hurt anything

**Comment by louis1706** [comment]
> Moving `Map.FindParentRoom` is a breaking change. I added these because Room.Get() isn't exactly as obvious. Having them doesn't hurt anything

I know it's breaking change it's why i don't want it to be moved for now but i thinks everything related to Room should be kept in Room

**Review by IRacle1 — APPROVED** [review]

**Review by louis1706 — APPROVED** [review]

**Pull request merged**